### PR TITLE
Fix exceptions serialization

### PR DESCRIPTION
**Pull request by hazzik**
This is a partial rework of #1469 in a non-breaking way concerning only exceptions.

### CODE REVIEW

File: src/NHibernate/PropertyNotFoundException.cs [hunk 1]
```diff
@@ -65,23 +59,41 @@ public PropertyNotFoundException(string propertyName, string fieldName, System.T
...
+				else if (entry.Name == "AccessorType")
+				{
+					AccessorType = entry.Value?.ToString();
+				}
+				else if (entry.Name == "TargetType")
+				{
+					var typeName = entry.Value?.ToString();
+					if (!string.IsNullOrEmpty(typeName))
+						TargetType = System.Type.GetType(typeName, true);
+				}
```
Comment (fredericDelaporte) [inline src/NHibernate/PropertyNotFoundException.cs]: When having many values, a `switch` seems more appropriate.
```c#
switch (entry.Name)
{
	case "PropertyName":
		PropertyName = entry.Value?.ToString();
		break;
	case "AccessorType":
		AccessorType= entry.Value?.ToString();
		break;
	case "TargetType":
		var typeName = entry.Value?.ToString();
		if (!string.IsNullOrEmpty(typeName))
			TargetType = System.Type.GetType(typeName, true);
		break;
}
```
(`DuplicateMappingException` and `LazyInitializationException` have two values, others only one.)

File: src/NHibernate/SchemaValidationException.cs [hunk 2]
```diff
@@ -16,7 +18,7 @@ public SchemaValidationException(string msg, IList<string> validationErrors) : b
 
 		protected SchemaValidationException(SerializationInfo info, StreamingContext context) : base(info, context)
 		{
-			ValidationErrors =
+			ValidationErrors = 
 				(ReadOnlyCollection<string>) info.GetValue("ValidationErrors", typeof(ReadOnlyCollection<string>));
```
Comment (fredericDelaporte) [inline src/NHibernate/SchemaValidationException.cs]: Undue added space.

File: src/NHibernate/PropertyNotFoundException.cs (hunk 1 above)
Comment (hazzik) [reply src/NHibernate/PropertyNotFoundException.cs]: I did consider this option and even written it this way, but for some reasons decided against it.

File: src/NHibernate/SchemaValidationException.cs (hunk 2 above)
Comment (hazzik) [reply src/NHibernate/SchemaValidationException.cs]: Force-pushed with revert

**Pull request merged**